### ISSUE TITLE
Rails 3.2.22.1 actionpack-3.2.22.1/lib/action_view/template/resolver.rb:113 syntax error, unexpected ':', expecting kEND

### ISSUE THREAD

**Issue by cntrytwist** [rails/rails#23248]
This error comes up when I attempt to run: rails s
So, to troubleshoot, I created a brand new blank app. Yes, it starts up just fine. Next step was to create a controller and a view. So I added the hello_controller.rb with the index method. Yes, I did put the proper route in the routes file.
Now it generates the following error when I attempt to access that hello index method.
Is there something I need to change in my apps?
Thanks

programmer@dev01:~/programming/BloggerApp$ rails s -p 3002
=> Booting WEBrick
=> Rails 3.2.22.1 application starting in development on http://0.0.0.0:3002
=> Call with -d to detach
=> Ctrl-C to shutdown server
[2016-01-25 14:39:33] INFO  WEBrick 1.3.1
[2016-01-25 14:39:33] INFO  ruby 1.8.7 (2014-01-28) [x86_64-linux]
[2016-01-25 14:39:33] INFO  WEBrick::HTTPServer#start: pid=4931 port=3002
[2016-01-25 14:39:39] ERROR SyntaxError: /home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_view/template/resolver.rb:113: syntax error, unexpected ':', expecting kEND
...ternal_files, instance_reader: false, instance_writer: false
                              ^
/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_view/template/resolver.rb:113: Can't assign to false
...files, instance_reader: false, instance_writer: false
                              ^
/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_view/template/resolver.rb:113: syntax error, unexpected ':', expecting '='
...eader: false, instance_writer: false
                              ^
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_view/railtie.rb:50
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/lazy_load_hooks.rb:36:in `instance_eval'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/lazy_load_hooks.rb:36:in `execute_hook'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/lazy_load_hooks.rb:43:in `run_load_hooks'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/lazy_load_hooks.rb:42:in `each'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/lazy_load_hooks.rb:42:in `run_load_hooks'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_view/base.rb:218
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/dependencies.rb:251:in `require'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/dependencies.rb:251:in `require'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/dependencies.rb:236:in `load_dependency'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/activesupport-3.2.22.1/lib/active_support/dependencies.rb:251:in `require'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/abstract_controller/view_paths.rb:1
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/abstract_controller/rendering.rb:35
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/abstract_controller/layouts.rb:195
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_controller/base.rb:183
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_dispatch/middleware/static.rb:32:in `ext'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_dispatch/middleware/static.rb:16:in `match?'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/actionpack-3.2.22.1/lib/action_dispatch/middleware/static.rb:77:in `call'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/engine.rb:484:in `call'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/application.rb:231:in `call'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/rack-1.4.7/lib/rack/content_length.rb:14:in `call'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/rack/log_tailer.rb:17:in `call'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/rack-1.4.7/lib/rack/handler/webrick.rb:59:in `service'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/httpserver.rb:104:in `service'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/httpserver.rb:65:in `run'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:173:in `start_thread'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:162:in `start'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:162:in `start_thread'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:95:in `start'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:92:in `each'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:92:in `start'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:23:in `start'
	/home/programmer/.rvm/rubies/ruby-1.8.7-head/lib/ruby/1.8/webrick/server.rb:82:in `start'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/rack-1.4.7/lib/rack/handler/webrick.rb:13:in `run'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/rack-1.4.7/lib/rack/server.rb:268:in `start'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/commands/server.rb:70:in `start'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/commands.rb:55
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/commands.rb:50:in `tap'
	/home/programmer/.rvm/gems/ruby-1.8.7-head@rails3/gems/railties-3.2.22.1/lib/rails/commands.rb:50
	script/rails:6:in `require'
	script/rails:6

**Comment by simi** [comment]
https://github.com/rails/rails/commit/18269d250fa58001ce7d8318571546aa90412975#commitcomment-15662448

**Comment by maclover7** [comment]
Yikes, not sure if this falls under our maintenance policy. 3-2-stable only receives security fixes, but not bug fixes. Bug fixes related to security fixes is a gray area for me... cc @tenderlove @rafaelfranca

**Comment by rjhancock** [comment]
I wouldn't call it a regression but as it's a bug fix to a security patch I think it falls under a need to fix.

I have a legacy app that I'm desperately trying to get it up to a newer version of Ruby but due to Unicode requirements having major issues between how 1.8.7 and 1.9.3 handle it.

**Comment by tenderlove** [comment]
Yes, we will fix this and release a new patch level version.

**Comment by tenderlove** [comment]
Also, if someone could send a PR that fixes this on 1.8, I'd happily merge and cut a new release. Getting 1.8 installed on my machine is not easy.

**Comment by simi** [comment]
I'm on it.

**Comment by tenderlove** [comment]
@simi thank you! 🙇🙇🙇🙇🙇🙇

**Comment by tkrajcar** [comment]
thanks @simi and @tenderlove! Those of us down here still in the 'we still support 1.8' dungeon sure appreciate it :bow: :smiley_cat:

**Comment by cntrytwist** [comment]
@simi @rafaelfranca @tenderlove You all are the best. Thanks for your excellent and timely work.

**Comment by marutosi** [comment]
Please release 3.2.22.2.
http://www.redmine.org/builds/logs/build_2.6-stable_postgresql_ruby-1.8.7-p358_110.html

**Comment by maclover7** [comment]
@marutosi - this will be fixed in an upcoming release of 3.2.22.2, coming out soon.

**Comment by marutosi** [comment]
When Rails 3.2.22.2 will release?
This is very ugly.
https://github.com/redmine/redmine/commit/875f539195921a61fad88d2644a6f33358106734
```ruby
if RUBY_VERSION >= "1.9.3"
  gem "rails", "3.2.22.1"
else
  gem "rails", :github => "rails/rails", :branch => "3-2-stable"
end
```

**Comment by rafaelfranca** [comment]
The answer for that question is always "when it is ready". Sorry for not having a better answer but asking release dates is not productive at all. We try to release a stable version one time each month and for security releases is when it is ready. For this specific case the "when it is ready" is the rule that applies.

**Comment by marutosi** [comment]
But, this issue is **regression** of **security** release, right?

**Comment by rafaelfranca** [comment]
yes, right, but if it is not ready why should we release?

**Comment by marutosi** [comment]
What is bottle neck?

**Comment by rafaelfranca** [comment]
Time. Understand that we are doing our best to release it as soon as possible. You are not the only one that needs this released but we can't do anything apart of what is already is being done. Like I said asking when it will be released or asking why will not make it be released fast.

It will be released when it is done. Sorry if this answer is not enough for you but it is the only one that we can make.

**Comment by rjhancock** [comment]
It's a pain to wait but I'd rather you guys take your time and release it when it's ready than speed release it. Especially for a framework that is security only and a ruby version that is beyond obsolete.

Keep up the good work.

**Comment by simi** [comment]
@rafaelfranca actually I think 3.2.22.1 version is not usable so last working version is 3.2.22 and that means that publicly disclaimed security issue is still present . It will be really nice to release new version ASAP.

I can offer my time to help and prepare new release. Is there anything I can help with?

**Comment by rafaelfranca** [comment]
What do you mean with not usable? You mean not usable with a version of Ruby that doesn't have support for 2 years and half?

We are doing the release ASAP.

**Comment by rjhancock** [comment]
@simi it is usable on 1.9.3 and up. It just doesn't work on a version that hasn't been supported in over 2 years.

1.9.3 is also no longer being maintained.

In short, quit bitching because you're not getting top tier support for something that doesn't have support.

The bug fix will be released when it's ready. If you have an issue with it, fix your code and update it.

**Comment by simi** [comment]
You can't start web server :( That's what I mean by "not usable". I thought it is still supported for security fixes and since this is fix of security fix (and I know that's only for 1.8).

I know everyone is doing the best to release it ASAP. I just wrote that message primarily to offer my help if there's something I can help with and not to offend or blame someone. I'm sorry if that was interpreted in that way.

**Comment by rafaelfranca** [comment]
@simi don't worry we got your message correctly. I was just trying to say that we are doing our best and also pointing that it is usable if you are using supported versions of Ruby. If there was something people could help I'd gladly point but there is not, now it is up to security team to deal with the release.
